solutions/usaco-899.mdx

```diff
@@ -0,0 +1,114 @@
+---
+id: usaco-899
+source: USACO Gold 2019 January
+title: Shortcut
+author: Ryan Chou
+---
+
+[Official Analysis (C++)](http://www.usaco.org/current/data/sol_shortcut_gold_jan19.html)
+
+## Explanation
+
+A simple brute-force approach would be to find the shortest path for all of the nodes, and then try to add a shortcut from $1$ to $i$ and find the shortest path again for all $i$.
+
+Using this approach, the answer is $\min(\texttt{dist}-\texttt{short}_i)$ for all i, where $\texttt{dist}$ is the travel time that all cows take without a shortcut and $\texttt{short}_i$ represents the travel time that all cows take when there is a shortcut from $1$ to $i$. Although this will certainly TLE, it hints at a faster way to compute the time saved. 
+
+Let's call $\texttt{occ}_i$ the number of cows that pass through field $i$ and $\texttt{trav}_i$ the cost for one cow to get from field $i$ to $1$. 
+Then the answer would be $\max(\texttt{occ}_i \cdot (\texttt{trav}_i - T))$ for all i. This works because $\texttt{trav}_i - T$ calculates the decrease in travel time for one cow to reach field $1$. By multiplying this value by $\texttt{occ}_i$, we calculate the time decrease for all the cows that pass through that field. 
```

Comment:
```suggestion
Then the answer would be $\max(\texttt{occ}_i \cdot (\texttt{trav}_i - T))$ for all $i$. This works because $\texttt{trav}_i - T$ calculates the decrease in travel time for one cow to reach field $1$. By multiplying this value by $\texttt{occ}_i$, we calculate the time decrease for all the cows that pass through that field. 
```